src/role_carry.js

```diff
@@ -164,6 +192,35 @@ roles.carry.preMove = function(creep, directions) {
   }
 
   reverse = creep.pickupWhileMoving(reverse);
+
+  // recycle carry
```

Comment:
I do not like the recycling idea sooo much.

If there is a problem with too many carries, the `call carry` logic should be improved.
But fine until the `call carry` logic is better.

Reply:
The ultimate solution to `call carry` logic would be #294.  My solution is making use of 'negative feedback'.  The latter one is not as efficient as the former one, but it's easier to be implemented, and, in my opinion, their performance difference is insignificant.

Reply:
I'm fine with adding this one and I'm also not so sure about the perfect solution, especially due to the fact that paths should be shared over multiple sources, which makes it more tricky.